### DISCUSSION TITLE
How to disable keymaps from 'lazyvim.config.keymaps'

### DISCUSSION THREAD

**Discussion by ghry5** [LazyVim/LazyVim#103] [Q&A] [answered]
I have the folling file `lua/plugins/navigator.lua`
```lua
return {
  {
    "numToStr/Navigator.nvim",
    config = function()
      require("Navigator").setup()
    end,
    keys = {
      { "<C-h>", "<cmd>NavigatorLeft<cr>", desc = "Navigate to left pane" },
      { "<C-l>", "<cmd>NavigatorRight<cr>", desc = "Navigate to right pane" },
      { "<C-j>", "<cmd>NavigatorDown<cr>", desc = "Navigate to bottom pane" },
      { "<C-k>", "<cmd>NavigatorUp<cr>", desc = "Navigate to top pane" },
    },
  },
}
```
However these keymaps are never applied. I suspect the default keymaps provided in `lazyvim.config.keymaps` are overriding them. I tried turning them off in `lua/config/lazy.lua`
```lua
require("lazy").setup({
  spec = {
    -- import LazyVim plugins
    {
      "LazyVim/LazyVim",
      import = "lazyvim.plugins",
      keys = {
        { "<C-h>", false },
        { "<C-l>", false },
        { "<C-j>", false },
        { "<C-k>", false },
      },
    },
    ...
```
However it doesn't seam to work.

**Comment by folke** [comment, marked as answer]
I just pushed a fix for this. We now no longer create keymaps if a lazy keys handler exists for th keymap.
https://github.com/LazyVim/LazyVim/commit/012dbab586529484cfec4d1c399b621a01e4164e

**Comment by stoopkidddd** [comment]
As far as I can tell I am on the latest version and seem to still be having the same problem, but I very well might be doing something wrong. 

I have the following file `lua/plugins/hop.lua` with the contents below:

```
return {
  "phaazon/hop.nvim",
  config = function() 
    require("hop").setup()
  end,
  keys = { 
    { "<C-f>", "<cmd>HopWord<cr>", desc = "Hop to Word" },
  },
}
```

I confirmed that if I set the key to something unused, for instance `<leader>hw`, it works as expected.

**Reply by folke** [reply]
Noice overrides that keymap, so you'll have to add:

```lua
  {
    "folke/noice.nvim",
    keys = {
      {"<c-f>", false},
    }
  }
```
to disable that

**Reply by stoopkidddd** [reply]
Ah okay I misunderstood the original response I think. So you still need to disable the original keymap in order to override? I was assuming it was automatic. At any rate, thanks for the response! This whole thing is awesome, much appreciated all your work =)

**Reply by gengdz** [reply]
> Noice overrides that keymap, so you'll have to add:
> 
> ```lua
>   {
>     "folke/noice.nvim",
>     keys = {
>       {"<c-f>", false},
>     }
>   }
> ```
> 
> to disable that

@folke hello~, I tried to do this in `lua/plugins/ui`, but it doesn't work. what should i do?

**Reply by folke** [reply]
@gengdz can you post your full ui.lua?

**Reply by gengdz** [reply]
```lua
return {
  {
    "folke/noice.nvim",
    keys = {
      { "<C-f>", false },
      { "<C-b>", false },
    },
  },
}

```
![image](https://user-images.githubusercontent.com/41912568/217313702-2e4713dc-3228-4451-9735-648e025778a9.png)

**Reply by gengdz** [reply]
@folke here is my nvim dotfiles.

**Comment by folke** [comment]
ok, I see it now, you need to add `mode = {"i", "n", "s"}` to the keymaps you want to disable.

So:

```lua
{ "<c-f>", false, mode = {"i", "n", "s"} },
```

**Reply by gengdz** [reply]
I I'm sorry to tell you, but it still doesn't work.

**Reply by folke** [reply]
```lua
  {
    "folke/noice.nvim",
    keys = {
      { "<c-f>", false, mode = { "i", "n", "s" } },
      { "<c-b>", false, mode = { "i", "n", "s" } },
    },
  },
```

Small letter `c` in `<c-...>`

**Reply by gengdz** [reply]
ooo, it works. awesome! really appreciate!
